prinz/src/main/scala/pl/touk/nussknacker/prinz/mlflow/model/RegisteredModel.scala

```diff
@@ -0,0 +1,16 @@
+package pl.touk.nussknacker.prinz.mlflow.model
+
+import pl.touk.nussknacker.prinz.model.Model
+
+case class ModelVersionTag(key: String, value: String)
+
+case class RegisteredModelTag(key: String, value: String)
+
+case class ModelVersion(name: String, version: String, creation_timestamp: String, last_updated_timestamp: String,
+                        current_stage: String, source: String,
+                        run_id: String, status: String, tags: List[ModelVersionTag])
+
+case class RegisteredModel(name: String, creation_timestamp: String, last_updated_timestamp: String,
+                           latest_versions: List[ModelVersion], tags: List[RegisteredModelTag]) extends Model {
+  override def getName(): String = name
+}
```

Comment:
make separated Model as the object that implements some abstract interface from RESTmodel of registered model in mlflow. You can create some helper function in object MlflowRestModel { def toMlflowModel: MlflowModel } or something  similar bout rather don't mix the REST api and received objects with objects that will have some functions to for example anylazy input data and give some predictions.